---
output:
  pdf_document:
    latex_engine: xelatex
  html_document:
    df_print: paged
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library("knitr")
library("dplyr")
library("tidyr")
library("kableExtra")
library("pander")
load("AiptasiaFigsAndTables.RData")
```

# Title Page

# Abstract

# Introduction

Anthozoans are a class of Cnidarians that includes scleractinian corals,
anemones and their relatives. Despite a global rise in disease
epizootics impacting scleractinian corals, very little is known about
the immune response of this key group of invertebrates. Initial
characterizations of immune related genes in cnidarians indicated that
they possess key components of the Toll/TLR pathway, complement C3,
membrane attack complex/perforin domains, and other components of innate
immunity once thought to have evolved much later [^Miller2007],
[^Nyholm2012], [^Shinzato2011],[^Putnam2007] and yet it was not
known whether these immune pathways were actually used in cnidarians to
mount a response against pathogens. A number of groups have since used
RNA-seq to produce some of the first profiles of anthozoan innate
immunity [^Vanderburg2016], [^Fuess2017], [^Weiss2013],
[^Poole2014], [^Libro2013]. To date, at least 9 studies have
profiled the immune response of corals and their anthozoan relatives,
and the data suggest that the immune response varies across anthozoans
and/or immune exposures. For example, Weiss et al. (2013) studied the
response of *Acropora millepora* to the bacterial cell wall derivative
muramyl dipeptide (MDP), finding up-regulation of GTPases of immunity
associated proteins (GiMAPs) [^Weiss2013]. GiMAPS were formerly
primarily associated with vertebrates and plants [^Liu2008]
[^Wang2009]. Vidal-Dupiol et al. (2014) compared the transcriptomic
responses of reef coral *Pocillopora damicornis* to thermal stress and
*Vibrio coralliilyticus* infection and observed that immune pathways -
including like Toll/TLR, complement, prophenoloxidase, and the
leukotriene cascade pathways – were up-regulated due to exposure to V.
coralliilyticus. Libro et al. (2013) compared the immune response of
healthy and diseased *Acropora cervicornis* coral using RNA-seq and
found that C-type lectins, ROS production, arachidonic acid metabolism,
and allene oxide production were strongly up-regulated in diseased
corals [^Libro2013]. Up-regulation of C-type lectins and ROS
production are hallmarks of phagocytosis, and the metabolism of
arachidonic acid via the allene oxide pathway has been linked to
eicosanoid synthesis in wounded corals [^Lõhelaid]. Interestingly,
Libro et al. (2013) did not identify strong up-regulation of genes
associated with the classic innate immune pathways such as Toll-like
receptor pathway or Prophenoloxidase pathway. The lack of unity in
response may reflect differences in annotation strategies or differences
in pathway versus gene-level analyses.

Reef-building corals and other anthozoans like *Aiptasia* are well known
for their symbiotic relationship with the dinoflagellate *Symbiodinium*
(also called zooxanthellae or “zoox”). This presents a challenge with
regard to the immune system because both pathogens and symbionts are
non-self, but pathogens need to be eliminated while symbionts are
allowed to remain. The symbionts live in vacuoles inside the endodermis
of the coral and give the host up to 95% of its energy as translocated
polysaccharides. Symbiosis requires clear communication between the host
and symbiont. During the establishment of symbiosis, the host must be
able to recognize symbionts, engulf them in phagosomes, and shield these
phagosomes from destruction [^Davy2012]. This suggests a clear link
between symbioses and immunity wherein symbionts evade the immune
response. Arrest of phagosomal maturation by Rab GTPases [^Davy2012]
and suppression of immune responses by transforming growth factor beta
(TGF$\beta$) have been identified as potential mechanisms by which symbionts
are shielded from destruction by the immune system [^Detournay2012].
Once symbiosis is established, the host must regulate the growth of the
symbionts and eject dead or dying symbionts [^Davy2012]. Regulation of
nutrients has been identified as one mechanism by which the host can
prevent overgrowth of the dinoflagellates [^Davy2012]. For corals and
other symbiotic anemones, we know very little about how the symbiosis is
maintained. Instead, we know a great deal about the breakdown in the
symbiosis – known as bleaching – when the symbionts are degraded or
expelled by the coral host under stressful conditions like heat stress,
cold stress, excessive UV exposure, pollutants, and disease. In addition
to naturally occurring stressors, chemical bleaching agents have been
identified including menthol and photosynthesis inhibitors. Several
mechanisms have been identified in the degradation and expulsion of the
symbionts, and the mechanisms vary depending on the type of stress.
These mechanisms include apoptosis, necrosis, and symbiont digestion via
autophagy (symbiophagy) [^Dani2016]. Apoptosis and necrosis
predominate in heat-stress bleaching, while symbiophagy predominates in
menthol bleaching. [^Dani2016] Arrest of phagosomal maturation is
required for establishment of symbiosis, and Dani et al (2016) suggest
that a reengagement of phagosomal maturation is involved in the
breakdown.

The symbiotic anemone *Exaiptasia pallida*, hereafter referred to as
*Aiptasia,* has become a model for studying symbiosis and immunity in
symbiotic anthozoans because (i) it is a hardy animal that can be made
aposymbiotic experimentally by exposing it to cold and heat stress
[Lehnert2014], as well as by treating it with compounds like menthol
[^Matthews2016], (ii) it can be propagated clonally [Sunagawa2009],
and (iii) a well-annotated genome for Aiptasia now exists
[Baumgarten2015]. Lehnert et al (2014) used RNA-seq in a
transcriptomic comparison of symbiotic and aposymbiotic anemones and
found about 900 differentially expressed genes involved in metabolite
transport, lipid metabolism, and amino acid metabolism. The immune
response of Aiptasia has recently been profiled; for example, Poole et
al (2016) used qPCR to compare complement activity in response to
colonization with Symbiodinium and the response to pathogen exposure
(*Serratia marcescens).* B-Factor 1 and MASP were up-regulated and
B-Factor 2b down-regulated in response to both pathogen exposure and
colonization. Matthews et al (2017) used RNA-Seq to profile immune and
nutrient exchange activity in response to colonization with
*Symbiodinium trenchii* and the response to colonization with the normal
symbiont, *Symbiodinium minutum*. The expression pattern after
colonization with the heterologous *S. trenchii* was intermediate
between the aposymbiotic state and the normal (*S. minutum*) symbiotic
state, with up-regulation of innate immune pathways in response to
heterologous colonization.

The immune response of *Aiptasia* can provide insight into the
evolutionary origin of the innate immune response as well as the
mechanism behind the breakdown of symbiosis that leads to bleaching in
corals. Transcriptome-wide comparisons have been made between symbiotic,
aposymbiotic, and heterologously colonized anemones [^Lehnert2014],
[^Matthews2017] and expression patterns of selected genes have been
profiled following pathogen exposure, but to-date, there has not been a
transcriptome-wide comparison of the response to immune challenge in
aposymbiotic versus aposymbiotic anemones.

In this study we explore the potential links between the maintenance of
the anthozoan-algal symbioses and immunity in a two-factor RNAseq
experiment comparing symbiotic versus aposymbiotic Aiptasia exposed to
the bacterial pathogen Vibrio coralliilyticus. Menthol-bleaching as the
method of producing aposymbiotic anemones because this method has a
lower mortality rate and produces generally healthier anemones, and
because the likely mechanism of menthol bleaching is activation of
autophagic digestion of Symbiodinium cells (symbiophagy) as part of host
innate immunity [^Dani2016]. Aposymbiotic anemones were produced using
a 24-hour menthol bleaching cycle. Both symbiotic and aposymbiotic
anemones were exposed to either live Vibrio or control for a total of
four treatment groups with six biological replicates per group.

# Methods

Wild anemones were obtained from Carolina Biological Supply. These
anemones, collected off the coast of North Carolina and kept in holding
tanks [^Carolina2015], are the source from which the widely-used cc7
clonal population was developed [^Sunagawa2009]. The wild anemones are
much larger than the available cc7 clones, and this facilitated
extraction of sufficient high-quality RNA without requiring pooling
within treatment groups. Anemones were maintained in culture plates and
to avoid any bias based on lighting intensity or other positional
effects, the wells of the plates were randomly assigned to six groups
using an R script. Thirty-six anemones were divided into bleached and
un-bleached. The bleached and un-bleached groups were then divided into
zoox counting, *Vibrio* treatment, and control groups for 6 anemones in
6 treatment groups. After plating and group assignment anemones were
maintained in the wells for a one-week acclimation phase. They were
exposed to a 12-hour day/night cycle with a light intensity of 70 µmol
quanta m<sup>-2</sup> s<sup>-1</sup>. To avoid contamination of the RNA
with any partially digested food the anemones were not fed during the
acclimation phase.

Aposymbiotic anemones were produced by exposure to 0.58 mM menthol/ASW
using a modified version of the protocol outlined by Wang, Chen, Tew,
Meng, and Chen (2012). The menthol exposure was on a 72-hour cycle, with
a 24-hour menthol exposure followed by a 48-hour resting period in FNSW.
A total of 36 anemones were put through the bleaching cycle. During the
bleaching cycle, 18 anemones had water replaced with .58 mM
menthol/FNSW, and 18 had water replaced with fresh FNSW.

The degree of bleaching was measured by homogenizing anemones with
BioMasher mortar and pestle sets in 1 mL FNSW, counting zoox with a
hemocytometer, and normalizing by animal weight. One homogenate tube was
dropped, and the homogenate lost, leaving 6 controls and 5 menthol
samples for counting. The homogenate was viewed using a hemocytometer.
Images were captured and *Symbiodinium* cells were manually counted
using ImageJ. The images showing manual counts are available under
supplemental figures. The menthol-bleached anemones had a 22-fold
reduction in *Symbiodinium* cells/mg wet weight. (df=1, f=62.94,
p=2.37e-05, control mean 20,909 cells/mg, bleached mean 936 cells/mg).

The live *Vibrio* groups were inoculated at a concentration of
10<sup>8</sup> CFU ml-1 in FNSW. The *Vibrio* inoculate was produced by
centrifuging marine broth cultures, drawing off the broth, and
re-suspending the pellet in FNSW. The anemones were exposed to either
*Vibrio* or control for 24 hours, then immediately homogenized in
Tri-Reagent.

The anemones were homogenized using BioMasher mortar and pestle sets.
Each anemone was first homogenized in 900 uL Tri-Reagent, then the 900
uL was divided into three separate tubes to which an additional 600 uL
Tri-Reagent was added to ensure sufficient volume to completely lyse the
cells. RNA was isolated separately from each of the 72 homogenate tubes
per the Tri-Reagent manufacturer’s protocol. The total RNA was analyzed
on an Agilent BioAnalyzer to obtain RIN scores.

For each of the 24 samples the RNA isolate with the highest RIN score
(mean score 6.88) was selected to proceed to mRNA isolation and library
preparation. mRNA was isolated using the NEBNext® Poly(A) mRNA Magnetic
Isolation Module and Illumina libraries were produced using the NEBNext®
Ultra™ Directional RNA Library Prep Kit for Illumina®. Multiplexed
paired-end libraries were sequenced on an Illumina HiSeq 2500 platform
at the FAS Center for System Biology at Harvard University. The reads
were adapter and quality-trimmed using Trimmomatic version 0.36
[^Bolger2014] using a 4-base sliding-window quality cutoff of 30
(Phred+33) and the TruSeq3 adapter sequence file (TruSeq3-PE.fa).

To facilitate KEGG pathway analysis, the transcript to gene annotation
file was mapped from *Aiptasia* predicted protein IDs to KEGG ortholog
IDs. Predicted coding sequences were extracted from the Aiptasia genome
annotation file using the gffread utility from the Cufflinks package
yielding 26,042 sequences. [^Trapnell2010]. The FASTA file produced by
gffread was aligned to Swiss-Prot [^UniProt2017] using blastx
[^NCBI2017]. Swiss-Prot hits were filtered using an e-value cutoff of
1e-5 and a minimum query coverage of 50%. Where species hits mapped to
multiple KEGG orthologs, the ortholog with the greatest number of
species hits was selected. The filtering and mapping was accomplished
with a custom Perl script, yielding 3,873 KEGG ortholog mappings.
Transcript counts were quantified against the predicted coding sequences
using Salmon [^Patro2017] and the transcript counts were imported into
DESeq2 [^Love2014] using tximport [^Soneson2015].

PERMANOVA and MDS analyses were used to identify transcriptome-wide
differences in gene expression due to menthol or vibrio exposure.
Hellinger-transformed DESeq2-normalized counts were analyzed using
PERMANOVA to identify transcriptome-wide differences in expression
patterns using the R package vegan [^Oksanen2018.tex]. MDS was used to
visualize transcriptome-wide differences between groups using the R
package metagenomeSeq []. Two-factor negative binomial GLM implemented
in DESeq2[] was used to identify differentially expressed (DE) genes
that differed due to menthol and vibrio exposure as well as the
interaction. The R package GAGE [] was used to identify KEGG pathways
that showed significant enrichment due to menthol or vibrio exposure.

[not sure where to put this yet] We performed two-way ANOVA on the
counts of trimmed and aligned reads and found no significant differences
based on Vibrio or Menthol exposure.

# Results

RNAseq data was produced for 24 anemones. There were 6 replicates each
for the treatment groups *Vibrio*/no menthol (symbiotic), no *Vibrio*/no
menthol (symbiotic), *Vibrio*/menthol (aposymbiotic), and no
*Vibrio*/menthol (aposymbiotic). Menthol treatment resulted in a
significant reduction in *Symbiodinium* densities (df=1, F=18.13,
p=.002) with a 22-fold reduction in algal symbionts in the menthol
treated “aposymbiotic” anemones. Menthol treated anemones were almost
completely white, but otherwise remained healthy-looking. None of the
anemones died after *Vibrio* exposure, and there were no visible
differences. The number of mapped RNA seq reads per anemone averaged
7,681,952 read pairs (+- 671,717 SE) with no mapping significant
differences due to either the menthol or *Vibrio* treatment
(supplementary table 1, two-way anova table).

**Multivariate**

Multivariate PERMANOVA analyses indicated strong and significant
differences in transcriptome-wide gene expression patterns (table 1) due
to Vibrio (p = 0.005) and menthol exposure (p = 0.001), but not the
Vibrio-menthol interaction. Vibrio explained 19.8% of the variation and
menthol explained 9.8% of the variation. Multi-dimensional scaling (MDS)
plots (figure 1) show these strong differences with Vibrio exposure
separated primarily on axis 1 and menthol exposure separated along axis
2.

**Univariate**

Univariate, negative binomial GLMs in DESeq2 [] were used to identify
genes that were differentially expressed due to menthol, Vibrio
exposure, and/or the interaction (figure 2; supplemental table 2). In
all, 4,164 genes were significantly DE for *Vibrio* exposure, 1,114
genes were DE for menthol exposure, and 66 genes were DE for the
*Vibrio*-menthol interaction. When you consider only genes with strong
protein annotations (Blast 50% query coverage and e-value 1e-5), there
were 1,038 DE annotated genes for Vibrio exposure, 401 DE annotated
genes for menthol exposure, and 10 DE annotated genes for the
Vibrio-menthol interaction. For visualization of sets and intersections
we used the UpSetR [] package. Venn and Euler diagrams work well for
showing sets and intersections for up to 3 or 4 sets, but they are
unable to effectively visualize intersections for larger numbers of set
intersections. For this reason, we used the plots produced by UpSetR,
which we will refer to as “UpSet” [] plots. The UpSet plot (fig. 2)
shows that for both Vibrio and menthol there were more genes
up-regulated than down-regulated, but this difference was much more
pronounced for *Vibrio* than for menthol. Figure 3 shows the counts of
up and down-regulated genes by treatment group. Vibrio alone showed the
strongest effect, with more up-regulated than down-regulated, followed
by menthol alone, with slightly more down-regulated than up-regulated.
The third major grouping consisted of 230 genes up for both *Vibrio* and
menthol, and 142 genes down for both *Vibrio* and menthol. 30 genes were
up for *Vibrio* and down for menthol. There were very few interactions,
with all remaining groupings consisting of 21 or fewer genes.

To facilitate the analysis of this large number of differentially
expressed genes, we performed KEGG pathway enrichment analysis using
GAGE [^Luo2009]. Twenty-three pathways were enriched due to Vibrio or
menthol exposure (table x); x pathways were enriched for Vibrio only, x
pathways were enriched for both Vibrio and Menthol, and no pathways were
enriched for only menthol. Pathways that were enriched for vibrio and
menthol included ribosome, phagosome, lysosome, and plant-pathogen
interaction, which are involved in x, x, and x, respectively. The 19
pathways enriched for *Vibrio* include five associated with immune
system (list them), four in signal transduction (x), three in endocrine
system (), two in cell growth and death (x), two in nervous system (x),
one in cell motility (), one in cellular community (), and one in
development ().

WE NEED TO MAKE ENRICHMENT AND DE INTO ONE TABLE MAYBE. THEN CHOSE TO
SUMMARIZE DE GENES IN PATHWAYS FOR KEY PATHWAYS – E.G. PHAGOSOME AND
LYSOSOME FOR BOTH. THEN, IMMUNE PATHWAYS FOR VIBRIO ETC. SOME OF WHAT
YOU HAVE IN DISCUSSION ABOUT PATHWAYS NEED TO MORE UP HERE. SOMETHING
LIKE THIS FROM DISCUSSION. As a percentage of annotated genes within the
pathway, the top five pathways up for Vibrio were complement and
coagulation cascades, TNF signaling pathway, Leukocyte transendothelial
migration, Toll-like receptor signaling pathway, and Osteoclast
differentiation. The top five pathways up for menthol were complement
and coagulation cascades, p53 signaling pathway, focal adhesion, TNF
signaling pathway, and phagosome.

# Discussion

GIVE GENERAL OVERVIEW. MULTIVARIATE AND UNIVARIATE ANALYSES INDICATE
THAT TREATMENT WITH VIBRIO AND MENTHOL HAVE STONG AND SIGNIFICANT
IMPACTS ON GENE EXPRESSION THAT ARE INDEPENDENT OR ADDITIVE, BUT NOT
INTERACTIVE. Vibrio had the strongest impact on DE expressed genes (n =
x) followed by menthol treatment (x DE genes) and then the combinations
of Vibrio and menthol (x). Gene annotations and pathway analyses
demonstrate that Vibrio was enriched for 19 pathways – including
pathways involved in immunity such as x, x, and x. Vibro and menthol
treatment overlapped in enrichment for 4 pathways including pathways
involved in phagosome and lysosome. No pathways were enriched for only
menthol. The shared pathways support the idea that the phagosome and
lysosome are both involved in symbioses and immunity (??are they the
same direction), whereas the innate immune responses are mostly reserved
for vibrio exposures and not menthol.

Discuss how our results compare to other immune responses in general.
Vidal, Libro etc. Up-regulation of immune genes in response to *Vibrio*
exposure was observed in two of the three primary innate immune pathways
– Toll/TLR and complement, but not the prophenoloxidase (PO) pathway.
NOW DISCUSS GENERAL OVERLAP. These results differ from those observed
for corals exposed to white-band disease, where little to no
differential expression was observed in these pathways. Now

Now start to discuss key pathways.

[move to results] As a percentage of annotated genes within the
pathway, the top five pathways up for Vibrio were complement and
coagulation cascades, TNF signaling pathway, Leukocyte transendothelial
migration, Toll-like receptor signaling pathway, and Osteoclast
differentiation. The top five pathways up for menthol were complement
and coagulation cascades, p53 signaling pathway, focal adhesion, TNF
signaling pathway, and phagosome.

For a gene-level comparison to the *Acropora cervicornis* immune
response presented in Libro et al 2013, the UniProt annotations assigned
to the differentially expressed *cervicornis* genes were mapped to the
KEGG ortholog IDs used for *Aiptasia*. Two of the genes identified in
Libro et al 2013 were also up-regulated in *Aiptasia* in response to
*Vibrio* exposure. These are caspase 3 and NADPH oxidase 2.

Genes up-regulated in response to Vibrio exposure were also compared to
the 1400-gene proteomic analysis of Aiptasia immune priming
[^Brown2015.ris]. Only one gene was found in common,
fumarylacetoacetase, and they were differentially expressed in opposite
directions.

This lack of overlap could be a limitation of trying to compare
experiments using a variety of annotation methods and IDs.

{It would take about a day to run through the same pipeline as Aiptasia
with cervicornis if you think it would be worthwhile for this
paper.}

## Complement and Coagulation Cascades [MIGHT BE COOL TO POINT OUT HOW MANY COMPLEMENT GENES ARE IN AIPTASIA VERSUS OVERALL. 

The complement system mediates innate immunity and provides a
nonspecific defense mechanism against pathogens. The three pathways of
complement are the classical pathway, the lectin pathway, and the
alternative pathway. Of these three pathways, the *Aiptasia* response to
*Vibrio* was primarily within the lectin and alternative pathways. The
up-regulation of genes within the complement pathway for both Vibrio and
menthol is consistent with other studies of ???? WHICH CORALS the role
of complement in both symbiosis and immune challenge [^Poole2016.ris],
[^Palmer2012.ris]. Within the KEGG Coagulation and Complement Cascades
pathway (figure 11) VWF, MASP1/2, and A2M were up-regulated for both
*Vibrio* and menthol, while TFPI, C3, MCP, C4BP were up-regulated only
for *Vibrio*. MASP 1 and 2 are mannan-binding lectin serine proteases
involved in the lectin pathway of the complement cascade [^NCBI Gene].
SOME SORT OF CONCLUSION ABOUT COMPLEMENT FUNCTION.

## TNF signaling pathway

Tumor necrosis factor (TNF) is a cytokine involved in apoptosis,
inflammation, and immunity. Quistad et al 2014 [^Quistad2014.ris]
found that corals possess more putative TNF receptors than any organism
previously described, including humans, but did not test for their
function in anthozoans. In the TNF Signaling pathway (figure 12),
TNF/LTA, TnF, and Sele were up-regulated for both Vibrio and Menthol.
CASP3, p38, AP-1, CREB, PI3K, c-Jun, Jun, and Mmp15 were up-regulated
for only *Vibrio*. Vcam1 was up only for menthol. Now talk about what
this means in terms of function of pathways. Describe other papers
showing TNF.

## Leukocyte transendothelial migration

## Toll-like receptor signaling pathway

In the Toll-Like Receptor Signaling Pathway (figure 15) MyD88, PI3K,
TRAF6, p38, and AP-1 were up-regulated for Vibrio but not menthol. LBP
was up-regulated for menthol only.

## Osteoclast differentiation

## p53 signaling pathway

## focal adhesion

## TNF signaling pathway

## Phagosome

Symbiophagy has been identified as a mechanism of Cnidarian bleaching
[^Downs2009.nbib], and during bleaching the symbiont vacuolar membrane
becomes a digestive organelle digesting the symbiont. This process is
similar to the process by which invading microorganisms are eliminated,
so it isn’t surprising that the phagosome pathway is enriched for both
*Vibrio* and menthol exposure. Within the phagosome pathway (figure 13)
vATPase, Sec61, and LAMP were up-regulated for both *Vibrio* and
menthol. Stx7, Hrs, a2B1, a5B1, B1, iC3b, p22phox, and p47phox. MARCO
was down-regulated for Vibrio only, CALR was down-regulated for moutput:
  pdf_document:
    latex_engine: xelatex
  html_document:
    df_print: paged
---
# Title Page

# Abstract

# Introduction

Anthozoans are a class of Cnidarians that includes scleractinian corals,
anemones and their relatives. Despite a global rise in disease
epizootics impacting scleractinian corals, very little is known about
the immune response of this key group of invertebrates. Initial
characterizations of immune related genes in cnidarians indicated that
they possess key components of the Toll/TLR pathway, complement C3,
membrane attack complex/perforin domains, and other components of innate
immunity once thought to have evolved much later [^Miller2007],
[^Nyholm2012], [^Shinzato2011],[^Putnam2007] and yet it was not
known whether these immune pathways were actually used in cnidarians to
mount a response against pathogens. A number of groups have since used
RNA-seq to produce some of the first profiles of anthozoan innate
immunity [^Vanderburg2016], [^Fuess2017], [^Weiss2013],
[^Poole2014], [^Libro2013]. To date, at least 9 studies have
profiled the immune response of corals and their anthozoan relatives,
and the data suggest that the immune response varies across anthozoans
and/or immune exposures. For example, Weiss et al. (2013) studied the
response of *Acropora millepora* to the bacterial cell wall derivative
muramyl dipeptide (MDP), finding up-regulation of GTPases of immunity
associated proteins (GiMAPs) [^Weiss2013]. GiMAPS were formerly
primarily associated with vertebrates and plants [^Liu2008]
[^Wang2009]. Vidal-Dupiol et al. (2014) compared the transcriptomic
responses of reef coral *Pocillopora damicornis* to thermal stress and
*Vibrio coralliilyticus* infection and observed that immune pathways -
including like Toll/TLR, complement, prophenoloxidase, and the
leukotriene cascade pathways – were up-regulated due to exposure to V.
coralliilyticus. Libro et al. (2013) compared the immune response of
healthy and diseased *Acropora cervicornis* coral using RNA-seq and
found that C-type lectins, ROS production, arachidonic acid metabolism,
and allene oxide production were strongly up-regulated in diseased
corals [^Libro2013]. Up-regulation of C-type lectins and ROS
production are hallmarks of phagocytosis, and the metabolism of
arachidonic acid via the allene oxide pathway has been linked to
eicosanoid synthesis in wounded corals [^Lõhelaid]. Interestingly,
Libro et al. (2013) did not identify strong up-regulation of genes
associated with the classic innate immune pathways such as Toll-like
receptor pathway or Prophenoloxidase pathway. The lack of unity in
response may reflect differences in annotation strategies or differences
in pathway versus gene-level analyses.

Reef-building corals and other anthozoans like *Aiptasia* are well known
for their symbiotic relationship with the dinoflagellate *Symbiodinium*
(also called zooxanthellae or “zoox”). This presents a challenge with
regard to the immune system because both pathogens and symbionts are
non-self, but pathogens need to be eliminated while symbionts are
allowed to remain. The symbionts live in vacuoles inside the endodermis
of the coral and give the host up to 95% of its energy as translocated
polysaccharides. Symbiosis requires clear communication between the host
and symbiont. During the establishment of symbiosis, the host must be
able to recognize symbionts, engulf them in phagosomes, and shield these
phagosomes from destruction [^Davy2012]. This suggests a clear link
between symbioses and immunity wherein symbionts evade the immune
response. Arrest of phagosomal maturation by Rab GTPases [^Davy2012]
and suppression of immune responses by transforming growth factor beta
(TGF$\beta$) have been identified as potential mechanisms by which symbionts
are shielded from destruction by the immune system [^Detournay2012].
Once symbiosis is established, the host must regulate the growth of the
symbionts and eject dead or dying symbionts [^Davy2012]. Regulation of
nutrients has been identified as one mechanism by which the host can
prevent overgrowth of the dinoflagellates [^Davy2012]. For corals and
other symbiotic anemones, we know very little about how the symbiosis is
maintained. Instead, we know a great deal about the breakdown in the
symbiosis – known as bleaching – when the symbionts are degraded or
expelled by the coral host under stressful conditions like heat stress,
cold stress, excessive UV exposure, pollutants, and disease. In addition
to naturally occurring stressors, chemical bleaching agents have been
identified including menthol and photosynthesis inhibitors. Several
mechanisms have been identified in the degradation and expulsion of the
symbionts, and the mechanisms vary depending on the type of stress.
These mechanisms include apoptosis, necrosis, and symbiont digestion via
autophagy (symbiophagy) [^Dani2016]. Apoptosis and necrosis
predominate in heat-stress bleaching, while symbiophagy predominates in
menthol bleaching. [^Dani2016] Arrest of phagosomal maturation is
required for establishment of symbiosis, and Dani et al (2016) suggest
that a reengagement of phagosomal maturation is involved in the
breakdown.

The symbiotic anemone *Exaiptasia pallida*, hereafter referred to as
*Aiptasia,* has become a model for studying symbiosis and immunity in
symbiotic anthozoans because (i) it is a hardy animal that can be made
aposymbiotic experimentally by exposing it to cold and heat stress
[Lehnert2014], as well as by treating it with compounds like menthol
[^Matthews2016], (ii) it can be propagated clonally [Sunagawa2009],
and (iii) a well-annotated genome for Aiptasia now exists
[Baumgarten2015]. Lehnert et al (2014) used RNA-seq in a
transcriptomic comparison of symbiotic and aposymbiotic anemones and
found about 900 differentially expressed genes involved in metabolite
transport, lipid metabolism, and amino acid metabolism. The immune
response of Aiptasia has recently been profiled; for example, Poole et
al (2016) used qPCR to compare complement activity in response to
colonization with Symbiodinium and the response to pathogen exposure
(*Serratia marcescens).* B-Factor 1 and MASP were up-regulated and
B-Factor 2b down-regulated in response to both pathogen exposure and
colonization. Matthews et al (2017) used RNA-Seq to profile immune and
nutrient exchange activity in response to colonization with
*Symbiodinium trenchii* and the response to colonization with the normal
symbiont, *Symbiodinium minutum*. The expression pattern after
colonization with the heterologous *S. trenchii* was intermediate
between the aposymbiotic state and the normal (*S. minutum*) symbiotic
state, with up-regulation of innate immune pathways in response to
heterologous colonization.

The immune response of *Aiptasia* can provide insight into the
evolutionary origin of the innate immune response as well as the
mechanism behind the breakdown of symbiosis that leads to bleaching in
corals. Transcriptome-wide comparisons have been made between symbiotic,
aposymbiotic, and heterologously colonized anemones [^Lehnert2014],
[^Matthews2017] and expression patterns of selected genes have been
profiled following pathogen exposure, but to-date, there has not been a
transcriptome-wide comparison of the response to immune challenge in
aposymbiotic versus aposymbiotic anemones.

In this study we explore the potential links between the maintenance of
the anthozoan-algal symbioses and immunity in a two-factor RNAseq
experiment comparing symbiotic versus aposymbiotic Aiptasia exposed to
the bacterial pathogen Vibrio coralliilyticus. Menthol-bleaching as the
method of producing aposymbiotic anemones because this method has a
lower mortality rate and produces generally healthier anemones, and
because the likely mechanism of menthol bleaching is activation of
autophagic digestion of Symbiodinium cells (symbiophagy) as part of host
innate immunity [^Dani2016]. Aposymbiotic anemones were produced using
a 24-hour menthol bleaching cycle. Both symbiotic and aposymbiotic
anemones were exposed to either live Vibrio or control for a total of
four treatment groups with six biological replicates per group.

# Methods

Wild anemones were obtained from Carolina Biological Supply. These
anemones, collected off the coast of North Carolina and kept in holding
tanks [^Carolina2015], are the source from which the widely-used cc7
clonal population was developed [^Sunagawa2009]. The wild anemones are
much larger than the available cc7 clones, and this facilitated
extraction of sufficient high-quality RNA without requiring pooling
within treatment groups. Anemones were maintained in culture plates and
to avoid any bias based on lighting intensity or other positional
effects, the wells of the plates were randomly assigned to six groups
using an R script. Thirty-six anemones were divided into bleached and
un-bleached. The bleached and un-bleached groups were then divided into
zoox counting, *Vibrio* treatment, and control groups for 6 anemones in
6 treatment groups. After plating and group assignment anemones were
maintained in the wells for a one-week acclimation phase. They were
exposed to a 12-hour day/night cycle with a light intensity of 70 µmol
quanta m<sup>-2</sup> s<sup>-1</sup>. To avoid contamination of the RNA
with any partially digested food the anemones were not fed during the
acclimation phase.

Aposymbiotic anemones were produced by exposure to 0.58 mM menthol/ASW
using a modified version of the protocol outlined by Wang, Chen, Tew,
Meng, and Chen (2012). The menthol exposure was on a 72-hour cycle, with
a 24-hour menthol exposure followed by a 48-hour resting period in FNSW.
A total of 36 anemones were put through the bleaching cycle. During the
bleaching cycle, 18 anemones had water replaced with .58 mM
menthol/FNSW, and 18 had water replaced with fresh FNSW.

The degree of bleaching was measured by homogenizing anemones with
BioMasher mortar and pestle sets in 1 mL FNSW, counting zoox with a
hemocytometer, and normalizing by animal weight. One homogenate tube was
dropped, and the homogenate lost, leaving 6 controls and 5 menthol
samples for counting. The homogenate was viewed using a hemocytometer.
Images were captured and *Symbiodinium* cells were manually counted
using ImageJ. The images showing manual counts are available under
supplemental figures. The menthol-bleached anemones had a 22-fold
reduction in *Symbiodinium* cells/mg wet weight. (df=1, f=62.94,
p=2.37e-05, control mean 20,909 cells/mg, bleached mean 936 cells/mg).

The live *Vibrio* groups were inoculated at a concentration of
10<sup>8</sup> CFU ml-1 in FNSW. The *Vibrio* inoculate was produced by
centrifuging marine broth cultures, drawing off the broth, and
re-suspending the pellet in FNSW. The anemones were exposed to either
*Vibrio* or control for 24 hours, then immediately homogenized in
Tri-Reagent.

The anemones were homogenized using BioMasher mortar and pestle sets.
Each anemone was first homogenized in 900 uL Tri-Reagent, then the 900
uL was divided into three separate tubes to which an additional 600 uL
Tri-Reagent was added to ensure sufficient volume to completely lyse the
cells. RNA was isolated separately from each of the 72 homogenate tubes
per the Tri-Reagent manufacturer’s protocol. The total RNA was analyzed
on an Agilent BioAnalyzer to obtain RIN scores.

For each of the 24 samples the RNA isolate with the highest RIN score
(mean score 6.88) was selected to proceed to mRNA isolation and library
preparation. mRNA was isolated using the NEBNext® Poly(A) mRNA Magnetic
Isolation Module and Illumina libraries were produced using the NEBNext®
Ultra™ Directional RNA Library Prep Kit for Illumina®. Multiplexed
paired-end libraries were sequenced on an Illumina HiSeq 2500 platform
at the FAS Center for System Biology at Harvard University. The reads
were adapter and quality-trimmed using Trimmomatic version 0.36
[^Bolger2014] using a 4-base sliding-window quality cutoff of 30
(Phred+33) and the TruSeq3 adapter sequence file (TruSeq3-PE.fa).

To facilitate KEGG pathway analysis, the transcript to gene annotation
file was mapped from *Aiptasia* predicted protein IDs to KEGG ortholog
IDs. Predicted coding sequences were extracted from the Aiptasia genome
annotation file using the gffread utility from the Cufflinks package
yielding 26,042 sequences. [^Trapnell2010]. The FASTA file produced by
gffread was aligned to Swiss-Prot [^UniProt2017] using blastx
[^NCBI2017]. Swiss-Prot hits were filtered using an e-value cutoff of
1e-5 and a minimum query coverage of 50%. Where species hits mapped to
multiple KEGG orthologs, the ortholog with the greatest number of
species hits was selected. The filtering and mapping was accomplished
with a custom Perl script, yielding 3,873 KEGG ortholog mappings.
Transcript counts were quantified against the predicted coding sequences
using Salmon [^Patro2017] and the transcript counts were imported into
DESeq2 [^Love2014] using tximport [^Soneson2015].

PERMANOVA and MDS analyses were used to identify transcriptome-wide
differences in gene expression due to menthol or vibrio exposure.
Hellinger-transformed DESeq2-normalized counts were analyzed using
PERMANOVA to identify transcriptome-wide differences in expression
patterns using the R package vegan [^Oksanen2018.tex]. MDS was used to
visualize transcriptome-wide differences between groups using the R
package metagenomeSeq []. Two-factor negative binomial GLM implemented
in DESeq2[] was used to identify differentially expressed (DE) genes
that differed due to menthol and vibrio exposure as well as the
interaction. The R package GAGE [] was used to identify KEGG pathways
that showed significant enrichment due to menthol or vibrio exposure.

[not sure where to put this yet] We performed two-way ANOVA on the
counts of trimmed and aligned reads and found no significant differences
based on Vibrio or Menthol exposure.

# Results

RNAseq data was produced for 24 anemones. There were 6 replicates each
for the treatment groups *Vibrio*/no menthol (symbiotic), no *Vibrio*/no
menthol (symbiotic), *Vibrio*/menthol (aposymbiotic), and no
*Vibrio*/menthol (aposymbiotic). Menthol treatment resulted in a
significant reduction in *Symbiodinium* densities (df=1, F=18.13,
p=.002) with a 22-fold reduction in algal symbionts in the menthol
treated “aposymbiotic” anemones. Menthol treated anemones were almost
completely white, but otherwise remained healthy-looking. None of the
anemones died after *Vibrio* exposure, and there were no visible
differences. The number of mapped RNA seq reads per anemone averaged
7,681,952 read pairs (+- 671,717 SE) with no mapping significant
differences due to either the menthol or *Vibrio* treatment
(supplementary table 1, two-way anova table).

**Multivariate**

Multivariate PERMANOVA analyses indicated strong and significant
differences in transcriptome-wide gene expression patterns (table 1) due
to Vibrio (p = 0.005) and menthol exposure (p = 0.001), but not the
Vibrio-menthol interaction. Vibrio explained 19.8% of the variation and
menthol explained 9.8% of the variation. Multi-dimensional scaling (MDS)
plots (figure 1) show these strong differences with Vibrio exposure
separated primarily on axis 1 and menthol exposure separated along axis
2.

**Univariate**

Univariate, negative binomial GLMs in DESeq2 [] were used to identify
genes that were differentially expressed due to menthol, Vibrio
exposure, and/or the interaction (figure 2; supplemental table 2). In
all, 4,164 genes were significantly DE for *Vibrio* exposure, 1,114
genes were DE for menthol exposure, and 66 genes were DE for the
*Vibrio*-menthol interaction. When you consider only genes with strong
protein annotations (Blast 50% query coverage and e-value 1e-5), there
were 1,038 DE annotated genes for Vibrio exposure, 401 DE annotated
genes for menthol exposure, and 10 DE annotated genes for the
Vibrio-menthol interaction. For visualization of sets and intersections
we used the UpSetR [] package. Venn and Euler diagrams work well for
showing sets and intersections for up to 3 or 4 sets, but they are
unable to effectively visualize intersections for larger numbers of set
intersections. For this reason, we used the plots produced by UpSetR,
which we will refer to as “UpSet” [] plots. The UpSet plot (fig. 2)
shows that for both Vibrio and menthol there were more genes
up-regulated than down-regulated, but this difference was much more
pronounced for *Vibrio* than for menthol. Figure 3 shows the counts of
up and down-regulated genes by treatment group. Vibrio alone showed the
strongest effect, with more up-regulated than down-regulated, followed
by menthol alone, with slightly more down-regulated than up-regulated.
The third major grouping consisted of 230 genes up for both *Vibrio* and
menthol, and 142 genes down for both *Vibrio* and menthol. 30 genes were
up for *Vibrio* and down for menthol. There were very few interactions,
with all remaining groupings consisting of 21 or fewer genes.

To facilitate the analysis of this large number of differentially
expressed genes, we performed KEGG pathway enrichment analysis using
GAGE [^Luo2009]. Twenty-three pathways were enriched due to Vibrio or
menthol exposure (table x); x pathways were enriched for Vibrio only, x
pathways were enriched for both Vibrio and Menthol, and no pathways were
enriched for only menthol. Pathways that were enriched for vibrio and
menthol included ribosome, phagosome, lysosome, and plant-pathogen
interaction, which are involved in x, x, and x, respectively. The 19
pathways enriched for *Vibrio* include five associated with immune
system (list them), four in signal transduction (x), three in endocrine
system (), two in cell growth and death (x), two in nervous system (x),
one in cell motility (), one in cellular community (), and one in
development ().

WE NEED TO MAKE ENRICHMENT AND DE INTO ONE TABLE MAYBE. THEN CHOSE TO
SUMMARIZE DE GENES IN PATHWAYS FOR KEY PATHWAYS – E.G. PHAGOSOME AND
LYSOSOME FOR BOTH. THEN, IMMUNE PATHWAYS FOR VIBRIO ETC. SOME OF WHAT
YOU HAVE IN DISCUSSION ABOUT PATHWAYS NEED TO MORE UP HERE. SOMETHING
LIKE THIS FROM DISCUSSION. As a percentage of annotated genes within the
pathway, the top five pathways up for Vibrio were complement and
coagulation cascades, TNF signaling pathway, Leukocyte transendothelial
migration, Toll-like receptor signaling pathway, and Osteoclast
differentiation. The top five pathways up for menthol were complement
and coagulation cascades, p53 signaling pathway, focal adhesion, TNF
signaling pathway, and phagosome.

# Discussion

GIVE GENERAL OVERVIEW. MULTIVARIATE AND UNIVARIATE ANALYSES INDICATE
THAT TREATMENT WITH VIBRIO AND MENTHOL HAVE STONG AND SIGNIFICANT
IMPACTS ON GENE EXPRESSION THAT ARE INDEPENDENT OR ADDITIVE, BUT NOT
INTERACTIVE. Vibrio had the strongest impact on DE expressed genes (n =
x) followed by menthol treatment (x DE genes) and then the combinations
of Vibrio and menthol (x). Gene annotations and pathway analyses
demonstrate that Vibrio was enriched for 19 pathways – including
pathways involved in immunity such as x, x, and x. Vibro and menthol
treatment overlapped in enrichment for 4 pathways including pathways
involved in phagosome and lysosome. No pathways were enriched for only
menthol. The shared pathways support the idea that the phagosome and
lysosome are both involved in symbioses and immunity (??are they the
same direction), whereas the innate immune responses are mostly reserved
for vibrio exposures and not menthol.

Discuss how our results compare to other immune responses in general.
Vidal, Libro etc. Up-regulation of immune genes in response to *Vibrio*
exposure was observed in two of the three primary innate immune pathways
– Toll/TLR and complement, but not the prophenoloxidase (PO) pathway.
NOW DISCUSS GENERAL OVERLAP. These results differ from those observed
for corals exposed to white-band disease, where little to no
differential expression was observed in these pathways. Now

Now start to discuss key pathways.

[move to results] As a percentage of annotated genes within the
pathway, the top five pathways up for Vibrio were complement and
coagulation cascades, TNF signaling pathway, Leukocyte transendothelial
migration, Toll-like receptor signaling pathway, and Osteoclast
differentiation. The top five pathways up for menthol were complement
and coagulation cascades, p53 signaling pathway, focal adhesion, TNF
signaling pathway, and phagosome.

For a gene-level comparison to the *Acropora cervicornis* immune
response presented in Libro et al 2013, the UniProt annotations assigned
to the differentially expressed *cervicornis* genes were mapped to the
KEGG ortholog IDs used for *Aiptasia*. Two of the genes identified in
Libro et al 2013 were also up-regulated in *Aiptasia* in response to
*Vibrio* exposure. These are caspase 3 and NADPH oxidase 2.

Genes up-regulated in response to Vibrio exposure were also compared to
the 1400-gene proteomic analysis of Aiptasia immune priming
[^Brown2015.ris]. Only one gene was found in common,
fumarylacetoacetase, and they were differentially expressed in opposite
directions.

This lack of overlap could be a limitation of trying to compare
experiments using a variety of annotation methods and IDs.

{It would take about a day to run through the same pipeline as Aiptasia
with cervicornis if you think it would be worthwhile for this
paper.}

## Complement and Coagulation Cascades [MIGHT BE COOL TO POINT OUT HOW MANY COMPLEMENT GENES ARE IN AIPTASIA VERSUS OVERALL. 

The complement system mediates innate immunity and provides a
nonspecific defense mechanism against pathogens. The three pathways of
complement are the classical pathway, the lectin pathway, and the
alternative pathway. Of these three pathways, the *Aiptasia* response to
*Vibrio* was primarily within the lectin and alternative pathways. The
up-regulation of genes within the complement pathway for both Vibrio and
menthol is consistent with other studies of ???? WHICH CORALS the role
of complement in both symbiosis and immune challenge [^Poole2016.ris],
[^Palmer2012.ris]. Within the KEGG Coagulation and Complement Cascades
pathway (figure 11) VWF, MASP1/2, and A2M were up-regulated for both
*Vibrio* and menthol, while TFPI, C3, MCP, C4BP were up-regulated only
for *Vibrio*. MASP 1 and 2 are mannan-binding lectin serine proteases
involved in the lectin pathway of the complement cascade [^NCBI Gene].
SOME SORT OF CONCLUSION ABOUT COMPLEMENT FUNCTION.

## TNF signaling pathway

Tumor necrosis factor (TNF) is a cytokine involved in apoptosis,
inflammation, and immunity. Quistad et al 2014 [^Quistad2014.ris]
found that corals possess more putative TNF receptors than any organism
previously described, including humans, but did not test for their
function in anthozoans. In the TNF Signaling pathway (figure 12),
TNF/LTA, TnF, and Sele were up-regulated for both Vibrio and Menthol.
CASP3, p38, AP-1, CREB, PI3K, c-Jun, Jun, and Mmp15 were up-regulated
for only *Vibrio*. Vcam1 was up only for menthol. Now talk about what
this means in terms of function of pathways. Describe other papers
showing TNF.

## Leukocyte transendothelial migration

## Toll-like receptor signaling pathway

In the Toll-Like Receptor Signaling Pathway (figure 15) MyD88, PI3K,
TRAF6, p38, and AP-1 were up-regulated for Vibrio but not menthol. LBP
was up-regulated for menthol only.

## Osteoclast differentiation

## p53 signaling pathway

## focal adhesion

## TNF signaling pathway

## Phagosome

Symbiophagy has been identified as a mechanism of Cnidarian bleaching
[^Downs2009.nbib], and during bleaching the symbiont vacuolar membrane
becomes a digestive organelle digesting the symbiont. This process is
similar to the process by which invading microorganisms are eliminated,
so it isn’t surprising that the phagosome pathway is enriched for both
*Vibrio* and menthol exposure. Within the phagosome pathway (figure 13)
vATPase, Sec61, and LAMP were up-regulated for both *Vibrio* and
menthol. Stx7, Hrs, a2B1, a5B1, B1, iC3b, p22phox, and p47phox. MARCO
was down-regulated for Vibrio only, CALR was down-regulated for menthol
only, and TUBB and F-actin were up-regulated for menthol only.

## Suppression of Ribosomal Function

The Ribosome pathway was enriched for both Vibrio and menthol, but in
opposite directions. Ribosomal function was down-regulated for Vibrio
and up-regulated for menthol. Although at first glance this didn’t
appear to be immune related, Vyleta et al (2012) identified ribosomal
suppression as a trigger of innate immune function via the NLRP3 in WHAT
ORGANISM. [^Vyleta2012.ris]

# Tables

All tables are also available as .csv files in the GitHub repository.

enthol
only, and TUBB and F-actin were up-regulated for menthol only.

## Suppression of Ribosomal Function

The Ribosome pathway was enriched for both Vibrio and menthol, but in
opposite directions. Ribosomal function was down-regulated for Vibrio
and up-regulated for menthol. Although at first glance this didn’t
appear to be immune related, Vyleta et al (2012) identified ribosomal
suppression as a trigger of innate immune function via the NLRP3 in WHAT
ORGANISM. [^Vyleta2012.ris]

# Tables

All tables are also available as .csv files in the GitHub repository.


```{r}
perm <- read.csv(file="Tables/PERMANOVA.csv")
perm <- subset(perm, select=-c(SumsOfSqs))
perm[is.na(perm)] <- ''
kable(perm, caption="Multivariate PERMANOVA",col.names = c("Factor", "Df", "Mean Square", "F-value","R2", "p-value"))
```
