---
title: "Figures"
author: "Chuck Roesel"
date: "3/30/2018"
output:
  pdf_document: default
  word_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = F,
	message = FALSE,
	warning = FALSE
)
library(dplyr)
library(tidyr)
library(knitr)
library(kableExtra)
library(pander)
library(UpSetR)
library(pathview)
library(igraph)
library(DESeq2)
library(data.table)

suppressWarnings(suppressMessages(library(metagenomeSeq)))
suppressWarnings(suppressMessages(library(vegan)))
```
```{r message=FALSE, warning=FALSE}
load("AiptasiaFigsAndTables.RData")
#load("AiptasiaGeneLevelAnalysis.RData")
```

```{r}
kxToKo <- read.csv("kxToKo.csv", row.names=1)
head(kxToKo)
```

```{r}
resVibrioAllNoAnnot <- results(ddsAllNoAnnot, alpha=.05, name="Vibrio_Vibrio_vs_NoVibrio")
dfVibrioAllNoAnnot <- as.data.frame(resVibrioAllNoAnnot)
data <- dfVibrioAllNoAnnot
#data <- subset(data, padj < .05, select=c("log2FoldChange", "padj"))
data <- subset(data, select=c("log2FoldChange", "padj"))
dfVibrioKO <- merge(data, kxToKo, by.x=0, by.y="KX")
head(dfVibrioKO)
```
```{r}
dfVibrioKO2 <- subset(dfVibrioKO, e < 1e-10 & cov > .2 & padj < .05)
dfVibrioKO2$lenDiff <- abs(dfVibrioKO2$cov - 1)
head(dfVibrioKO2)
```

```{r}
pathway <- "ko04620 Toll-like receptor signaling pathway"
subset <- 'query'
id <- unlist(strsplit(pathway, " "))[1]
genes <- unlist(kegg.sets.ko[pathway], use.names=FALSE)
dfBothAnnotSearch <- subset(dfVibrioKO2, dfVibrioKO2$KO %in% genes)
head(dfBothAnnotSearch)
```
```{r}
DT_all <- as.data.table(dfBothAnnotSearch) 
head(DT_all)
```
```{r}
testKO <- DT_all[order(lenDiff, e)]
test2KO <- testKO[, head(.SD,1), by = .(KO)] 
test2KO_t <- test2KO[order(lenDiff, e)] 
test3KO <- test2KO[, head(.SD,1), by = .(Row.names)]
test3KO
```

```{r}

#dfBothAnnotSearch <-merge(best, dfBothAnnotSearch)
#dfBothAnnotSearch
rownames(test3KO) <- test3KO$KO
best <- as.data.frame(test3KO)
rownames(best) <- best$KO
write.csv(best, file="tollAll.csv")
best <- subset(best, select=c("log2FoldChange"))
best
```

```{r}
#write.csv(dfBothAnnotSearch, file="toll.csv")
#pathExprs <- subset(dfBothAnnotSearch, select=c("KO","log2FoldChange"))
#pathExprs[is.na(pathExprs)] <- 02
pv.out <- pathview(best, pathway.id = id, kegg.dir = 'keggImages/', 
                   species = "ko", out.suffix = subset,
                   keys.align = "y", kegg.native = T, match.data = T, 
                   multi.state = F, same.layer = T, map.symbol=T, na.col = "gray", mid =
                     list(gene = "white", cpd = "white"), low =
                     list(gene = "blue", cpd = "blue"))
write.csv(unlist(pv.out), file=paste0(pathway, '.csv'))
```


