---
title: "Aiptasia Immune Response"
output:
  pdf_document:
    fig_caption: true
  html_document:
    df_print: paged
  md_document:
    variant: gfm
    df_print: paged
  word_document: default
classoption: portrait
editor_options: 
  chunk_output_type: inline
bibliography: Aiptasia.bib  
---

# Title Page

# Abstract

# Introduction

Anthozoans are a class of Cnidarians that includes scleractinian corals,
anemones and their relatives. Despite a global rise in disease
epizootics impacting scleractinian corals, very little is known about
the immune response of this key group of invertebrates. Initial
characterizations of cnidarian immune genes indicated that they possess
key components of the Toll/TLR pathway, complement C3, membrane attack
complex/perforin domains, and other components of innate immunity once
thought to have evolved much later \[^Miller2007\], \[^Nyholm2012\],
\[^Shinzato2011\],\[^Putnam2007\] and yet it was not known whether these
immune pathways were actually used in cnidarians to mount a response
against pathogens. A number of groups have since used RNA-seq to produce
some of the first profiles of anthozoan innate immunity
\[^Vanderburg2016\], \[^Fuess2017\], \[^Weiss2013\], \[^Poole2014\],
\[^Libro2013\]. To date, at least nine studies have profiled the immune
response of corals and their anthozoan relatives, and the data suggest
that the immune response varies across anthozoans and/or immune
exposures. For example, Weiss et al. (2013) studied the response of the
reef coral *Acropora millepora* to the bacterial cell wall derivative
muramyl dipeptide (MDP) and observed the up-regulation of GTPases of
immunity associated proteins (GiMAPs) \[^Weiss2013\]. GiMAPS were
formerly primarily associated with vertebrates and plants \[^Liu2008\]
\[^Wang2009\]. Vidal-Dupiol et al. (2014) compared the transcriptomic
responses of reef coral *Pocillopora damicornis* to thermal stress and
*Vibrio coralliilyticus* infection and observed that immune pathways -
including Toll/TLR, complement, prophenoloxidase, and the leukotriene
cascade pathways – were up-regulated due to exposure to *V.
coralliilyticus*. Libro et al. (2013) compared the immune response of
healthy and White Band Diseased infected *Acropora cervicornis* coral
using RNA-seq and found that C-type lectins, ROS production, arachidonic
acid metabolism, and allene oxide production were strongly up-regulated
in diseased corals \[^Libro2013\]. Up-regulation of C-type lectins and
ROS production are hallmarks of phagocytosis, and the metabolism of
arachidonic acid via the allene oxide pathway has been linked to
eicosanoid synthesis in wounded corals \[^Lõhelaid\]. Interestingly,
Libro et al. (2013) did not identify strong up-regulation of genes
associated with the classic innate immune pathways such as Toll-like
receptor pathway or Prophenoloxidase pathway.

Reef-building corals and other anthozoans like *Aiptasia* are also well
known for their symbiotic relationship with the dinoflagellate
*Symbiodinium* (also called zooxanthellae or “zoox”). This presents a
challenge with regard to the immune system because both pathogens and
symbionts can elicit an allorecognition response with the difference
being that pathogens are typically eliminated while symbionts have
mechanisms that are allowed (allow them?) to coexist within the host.
The symbionts live in vacuoles inside the endodermis of the coral and
give the host up to 95% of its energy as translocated polysaccharides.
Symbiosis requires clear communication between the host and symbiont.
During the establishment of symbiosis, the host must be able to
recognize symbionts, engulf them in phagosomes, and shield these
phagosomes from destruction \[^Davy2012\]. This suggests a clear link
between symbioses and immunity wherein symbionts evade the immune
response. Arrest of phagosomal maturation by Rab GTPases \[^Davy2012\]
and suppression of immune responses by transforming growth factor beta
(TGFβ) have been identified as potential mechanisms by which symbionts
are shielded from destruction by the immune system \[^Detournay2012\].
Once symbiosis is established, the host must regulate the growth of the
symbionts and eject dead or dying symbionts \[^Davy2012\]. Regulation of
nutrients has been identified as one mechanism by which the host can
prevent overgrowth of the dinoflagellates \[^Davy2012\].

For symbiotic corals and anemones, we know much less about how the
symbiosis is maintained than the breakdowns in the symbiosis – known as
bleaching – when the symbionts are degraded or expelled by the coral
host due to factors like thermal stress, UV exposure, and disease. In
addition to naturally occurring stressors, chemical bleaching agents
have been identified including menthol and photosynthesis inhibitors.
Several mechanisms have been identified in the degradation and expulsion
of the symbionts, and the mechanisms vary depending on the type of
stress. These mechanisms include apoptosis, necrosis, and symbiont
digestion via autophagy (symbiophagy) \[^Dani2016\]. Apoptosis and
necrosis predominate in heat-stress bleaching, while symbiophagy
predominates in menthol bleaching. \[^Dani2016\] Arrest of phagosomal
maturation is required for establishment of symbiosis, and Dani et al
(2016) suggest that a re-engagement of phagosomal maturation is involved
in the breakdown.

The symbiotic anemone *Exaiptasia pallida*, hereafter referred to as
*Aiptasia,* has become a model for studying symbiosis and immunity in
symbiotic anthozoans because (i) it is a hardy animal that can be made
aposymbiotic experimentally by exposing it to cold and heat stress
\[^Lehnert2014\], as well as by treating it with compounds like menthol
\[^Matthews2016\], (ii) it can be propagated clonally \[Sunagawa2009\],
and (iii) a well-annotated genome for *Aiptasia* now exists
\[^Baumgarten2015\]. Lehnert et al (2014) used RNA-seq in a
transcriptomic comparison of symbiotic and aposymbiotic anemones and
found about 900 differentially expressed genes involved in metabolite
transport, lipid metabolism, and amino acid metabolism. The immune
response of *Aiptasia* has recently been profiled; for example, Poole et
al (2016) used qPCR to compare complement activity in response to
colonization with *Symbiodinium* and the response to pathogen exposure
(*Serratia marcescens*). B-Factor 1 and MASP were up-regulated and
B-Factor 2b down-regulated in response to both pathogen exposure and
colonization. Matthews et al (2017) used RNA-Seq to profile immune and
nutrient exchange activity in response to colonization with
*Symbiodinium trenchii* and the response to colonization with the normal
symbiont, *Symbiodinium minutum*. The expression pattern after
colonization with the heterologous *S. trenchii* was intermediate
between the aposymbiotic state and the normal (*S. minutum*) symbiotic
state, with up-regulation of innate immune pathways in response to
heterologous colonization.

In this study, we explore the genetic links between the anthozoan-algal
symbioses and immunity in a two-factor RNA-seq experiment using
menthol-bleached *Aiptasia* exposed to the bacterial pathogen *Vibrio
coralliilyticus*. Menthol-bleaching was used to compare symbiotic
(untreated) versus aposymbiotic (menthol treated) anemones where the
hypothesized mechanism of menthol bleaching is thought to be the
activation of autophagic digestion of *Symbiodinium* cells (symbiophagy)
as part of host innate immunity \[^Dani2016\]. The bacterial pathogen
*Vibrio coralliilyticus* was used to initiate the immune response of
Aiptasia. The two-factor design comparing *Vibrio* and menthol exposure
as factors allowed us to identify gene expression patterns that were due
to *Vibrio* and/or menthol exposure as well as any interactions between
the exposures.

Methods

Wild anemones were obtained from Carolina Biological Supply. These
anemones, collected off the coast of North Carolina and kept in holding
tanks \[^Carolina2015\], are the source from which the widely-used cc7
clonal population was developed \[^Sunagawa2009\]. Anemones were
maintained in culture plates and held under 24 watt t5 fluorescent
lights for 12-hour light cycle. To avoid any bias based on lighting
intensity or other positional effects, the wells of the plates were
randomly assigned to six groups using an R script. Thirty-six anemones
were divided into menthol-treatment and control. The menthol-treatment
and control groups were then divided into zoox counting (n = 12
anemones), *Vibrio* treatment (n = 12 anemones), and control groups for
6 anemones in 6 treatment groups. After plating and group assignment
anemones were maintained in the wells for a one-week acclimation phase.
They were exposed to a 12-hour day/night cycle with a light intensity of
70 µmol quanta m<sup>-2</sup> s<sup>-1.</sup> To avoid contamination of
the RNA with any partially digested food the anemones were not fed
during the acclimation phase.

Aposymbiotic anemones were produced by exposure to 0.58 mM menthol/ASW
using a modified version of the protocol outlined by Wang, Chen, Tew,
Meng, and Chen (2012). The menthol exposure was on a 72-hour cycle, with
a 24-hour menthol exposure followed by a 48-hour resting period in 0.2
μm filtered natural sea water (FNSW). A total of 36 anemones were put
through the menthol treatment cycle. During the menthol treatment cycle,
18 anemones had water replaced with 0.58 mM menthol/FNSW, and 18 had
water replaced with fresh FNSW.

The degree of menthol bleaching was measured by homogenizing anemones
with BioMasher mortar and pestle sets in 1 mL FNSW, counting zoox with a
hemocytometer, and normalizing by animal weight for 6 controls and 5
menthol-treated anemones. Symbiodinium cells were manually counted using
a hemocytometer from three replicate cells. Menthol-bleached anemones
had a 22.5-fold reduction in *Symbiodinium* cells/mg wet weight. (df=1,
f=62.94, p=2.37e-05), control mean 24,901 cells/mg (+- 7,000), bleached
mean 1,105 cells/mg (+- 297).

The live *Vibrio* groups were inoculated at a concentration of
10<sup>8</sup> CFU ml<sup>-1</sup> in 0.2 μm filtered natural seawater
(FNSW) . *Vibrio coralliilyticus* strain BAA-450™ from ATCC<sup>®</sup>
\[\] was used for the Vibrio treatment. The *Vibrio* inoculate was
produced by centrifuging marine broth cultures, drawing off the broth,
and re-suspending the pellet in FNSW. The anemones were exposed to
either *Vibrio* or control for 24 hours, then immediately homogenized in
Tri-Reagent..

After the menthol treatment and Vibrio exposures were complete, the
anemones were homogenized using BioMasher mortar and pestle sets for
total RNA extraction. Each anemone was first homogenized in 900 uL
Tri-Reagent, then the 900 uL was divided into three separate tubes to
which an additional 600 uL Tri-Reagent was added to ensure sufficient
volume to completely lyse the cells. Total RNA was isolated using the
Tri-Reagent manufacturer’s protocol. Total RNA was quantified on an
Agilent BioAnalyzer to obtain concentrations and RIN scores.

For each of the 24 samples the RNA isolate with the highest RIN score
(mean score 6.88) was selected to proceed to mRNA isolation and library
preparation. mRNA was isolated using the NEBNext® Poly(A) mRNA Magnetic
Isolation Module and Illumina libraries were produced using the NEBNext®
Ultra™ Directional RNA Library Prep Kit for Illumina®. Multiplexed
paired-end libraries were sequenced on an Illumina HiSeq 2500 platform
at the FAS Center for System Biology at Harvard University. The reads
were adapter and quality-trimmed using Trimmomatic version 0.36
\[^Bolger2014\] using a 4-base sliding-window quality cutoff of 30
(Phred+33) and the TruSeq3 adapter sequence file (TruSeq3-PE.fa).

Transcript counts were quantified against the predicted coding sequences
using Salmon \[^Patro2017\] and the transcript counts were imported into
DESeq2 \[^Love2014\] using tximport \[^Soneson2015\].

To facilitate KEGG pathway analysis, the transcripts were mapped from
*Aiptasia* predicted protein IDs to KEGG ortholog IDs. Predicted coding
sequences were extracted from the *Aiptasia* genome annotation file
using the gffread utility from the Cufflinks package yielding 26,042
sequences. \[^Trapnell2010\]. The FASTA file produced by gffread was
aligned to Swiss-Prot \[^UniProt2017\] using blastx \[^NCBI2017\].
Swiss-Prot hits were filtered using an e-value cutoff of 1e-10 and a
minimum query coverage of 50%. Where blastx hits mapped to multiple KEGG
orthologs, the ortholog with the query coverage closest to 100% was
selected. For multiple hits with identical coverage the lowest e-value
was chosen. This same process was then applied in the reverse direction
to eliminate duplicate *Aiptasia* to KEGG ortholog mappings. The
filtering and mapping was accomplished with custom Perl and R scripts,
yielding 4,807 one-to-one *Aiptasia* to KEGG ortholog mappings. This
mapping table is available under supplemental data as aipAnnot2.csv.

PERMANOVA and MDS analyses were used to identify transcriptome-wide
differences in gene expression due to menthol or *Vibrio* exposure.
Hellinger-transformed DESeq2-normalized counts were analyzed using
PERMANOVA to identify transcriptome-wide differences in expression
patterns using the R package vegan \[^Oksanen2018\] and MDS was used to
visualize transcriptome-wide differences between groups \[\]. A
two-factor negative binomial GLM implemented in DESeq2\[\] was used to
identify differentially expressed (DE) genes that differed due to
*Vibrio* and menthol exposure as well as the interaction. The R package
ESGEA \[\] was used to identify KEGG pathways that showed significant
enrichment due to *Vibrio* or menthol exposure, and the R package
goseq\[\] was used to identify pathways overrepresented in DE genes.

\[not sure where to put this yet\] We performed two-way ANOVA on the
counts of trimmed and aligned reads and found no significant differences
based on Vibrio or Menthol exposure.

# Results

RNA-seq data was produced for 24 anemones; 6 replicates each for the
treatment groups: 1) *Vibrio* treated /no menthol (symbiotic), no
*Vibrio*/no menthol (symbiotic), *Vibrio* treated/menthol treated
(aposymbiotic), and no *Vibrio*/menthol treated (aposymbiotic). Menthol
treatment resulted in a significant 22-fold reduction in *Symbiodinium*
densities (ANOVA df=1, F=18.13, p=0.002); menthol-treated anemones
appeared completely white in color and had *Symbiodinium* densities
averaging 1,105 cells/mg (+- 297) compared to the untreated symbiotic
anemones, which averaged 24,901 cells/mg (+- 7,000). None of the
anemones died after *Vibrio* exposure, and there were no visible
differences in appearance or behavior of the anemones between the
treatment groups. The number of mapped RNA seq reads per anemone
averaged 7,681,952 read pairs (+- 671,717 SE) with no significant
mapping differences due to either the menthol or *Vibrio* treatment
(2-way ANOVA, supplementary table 1)).

![](mdMedia/media/image4.png)

**Multivariate**

Multivariate PERMANOVA analyses identified strong and significant
differences in transcriptome-wide gene expression patterns (table 1) due
to *Vibrio* (p = 0.007) and menthol exposure (p = 0.001), but not the
*Vibrio*-menthol interaction. *Vibrio* explained 19.8% of the variation
and menthol explained 9.8% of the variation. Multidimensional scaling
(MDS) plots (figure 1) show these strong differences with Vibrio
exposure separated primarily on axis 1 and menthol exposure separated
along axis 2.

![](mdMedia/media/image6.png)

Figure 1

**Univariate**

Univariate, negative binomial GLMs in DESeq2 \[\] were used to identify
genes that were differentially expressed due to *Vibrio* exposure,
menthol exposure, and/or the interaction (figure 2; supplemental table
2). In all, 4,164 genes were significantly DE for *Vibrio* exposure,
1,114 genes were DE for menthol exposure, and 66 genes were DE for the
*Vibrio*-menthol interaction. When you consider only genes with strong
protein annotations (Blast 50% query coverage, e-value 1e-10, best 1:1
ortholog match), there were 1,338 DE annotated genes for *Vibrio*
exposure, 462 DE annotated genes for menthol exposure, and 11 DE
annotated genes for the *Vibrio*-menthol interaction.

The UpSet plot (fig. 2) shows more genes up-regulated than
down-regulated for both *Vibrio* and menthol. *Vibrio* alone showed the
strongest effect, with more DE genes up-regulated (V+ = 2,155) than
down-regulated (V- = 1,552), followed by menthol alone, with slightly
more down-regulated (M- = 353) than up-regulated (M+ = 322) DE genes.
The third major grouping consisted of 230 genes that were up-regulated
for both *Vibrio* and menthol (V+M+), and 142 genes down-regulated for
both *Vibrio* and menthol (V-M-). 30 genes were up-regulated for
*Vibrio*, but down-regulated for menthol. There were very few
interactions, with all remaining groupings consisting of 21 or fewer
genes.

![](mdMedia/media/image5.png)

*Figure 2*

To facilitate the analysis of this large number of differentially
expressed genes, we performed KEGG pathway enrichment analysis using
ESGEA \[\]. We categorized pathways by KEGG pathway class and as
significantly enriched for *Vibrio* only (V), menthol only (M) or both
(VM). To further narrow the focus to immune-related KEGG pathway
classes, we selected immune system, transport and catabolism, and cell
growth and death. Within these classes we identified 11 pathways that
were significantly enriched for *Vibrio* and/or menthol exposure (table
2; fig. 3). The majority of pathways (7) were enriched for both *Vibrio*
and menthol (VM), followed by *Vibrio* only (V) with four pathways. No
pathways were enriched for menthol only.

![](mdMedia/media/image8.png)

Table 2:

Differential gene expression patterns shown in Table 2 for all eleven
enriched pathways indicating numbers of annotated genes that are DE for
*Vibrio* only \[up (V+) or down (V-)\], Menthol only \[M+ or M-\], or
Both \[V+M+, V-M-\]. While all 11 pathways have more than 5 DE genes
overall, only complement had a significant over-representation of DE
genes and only for *Vibrio* (adj-p = .004); the adjusted p-value for
menthol was nearly significant with an adj-p equal to 0.153.

When we compare expression patterns of DE genes that are highly
differentially expressed genes (log<sub>2</sub> fold change \>|1|)
across all 11 pathways, 42 genes were highly expressed (table x). For
the immune pathways, complement and coagulation cascade pathway had the
most highly expressed DE genes with 10 DE genes; four genes are part of
the coagulation cascade and the remaining six genes are part of the
complement cascade. Out of the four coagulation genes, three were
up-regulated for both *Vibrio* and menthol (A2MG, TFPI1, VFW) while F13B
was up-regulated for *Vibrio* only. Out of the six complement genes,
five genes within the complement alternative pathway were up-regulated
for *Vibrio* only (MCP, DAF, CFAB, C3, CFAH). The remaining complement
gene PAI2 was down-regulated for both *Vibrio* and menthol and is
involved in negative regulation of apoptosis, fibrinolysis, and wound
healing.

NOD-like receptor pathway, which includes toll-like receptor pathway,
had 6 highly expressed DE genes all of which were DE for *Vibrio* only
\[3 up-regulated and 3 down-regulated\]. MyD88, TRAF3 and Bcl2 were
up-regulated, while TRAF2, RIPK2, and CASR were down-regulated due to
*Vibrio* exposure. MyD88, TRAF3, and RIPK2 activation lead to NF-kappa-β
activation, cytokine secretion and the inflammatory response while BCL2
inhibits caspases and suppresses apoptosis. TRAF2 Regulates activation
of NF-kappa-β and JNK and CASR regulates calcium homeostasis.

Chemokine signaling pathway had 3 highly expressed DE genes for *Vibrio*
only \[1 up; 2 down\]. STAT1 was up-regulated, while CXCR4 and GBB4 were
down-regulated. STAT1 is a signal transducer and transcription activator
that mediates cellular responses to interferons and cytokines. CXCR4
transduces a chemokine signal by increasing intracellular calcium ion
levels and enhancing MAPK1/MAPK3 activation. GBB4 is a G protein
involved in modulating transmembrane signaling systems.

Antigen had 2 highly expressed DE genes; PSME2 was up for *Vibrio* only
and RFXK down for menthol only. PSME2 is a proteasome activator involved
in antigen processing. RFXK activates transcription from class II MHC
promoters.

For the transport and catabolism pathways, the peroxisome pathway had 7
highly expressed DE genes. One gene was up for both (PAOX), 3 genes were
down for *Vibrio* only (SOX, LCFB, DHRS4), one gene was down for both
(GNPAT), one gene was up for menthol only (EASC) and one gene was down
for menthol only (BAAT). Peroxisomes are involved in resolution of
bacterial infection through innate immune signaling \[DiCara2017\].

Endocytosis pathway had 6 highly expressed DE genes. Three genes were up
for *Vibrio* only (HRS, HSE1, CLH), 2 genes were down for *Vibrio* only
(VPS4, PLD2), and one gene was down for menthol only (JUNO). Lysosome
pathway had 4 highly expressed DE genes. One gene was up for *Vibrio*
only (AP3D), one gene was down for *Vibrio* only (GALNS), one gene was
up for menthol only (PAG15), and one gene was down for menthol only
(BGLR).

For cell growth and death, Apoptosis pathway had 5 highly expressed DE
genes. Two genes - both Caspases - were up for Vibrio only (CASP7,
CASP9), one gene was down for Vibrio only (TBA), one was up for menthol
(P53) and one was down for menthol only (BIRC5).

# Discussion

Multivariate and univariate analyses of *Aiptasia* gene expression
demonstrated that exposure to live *Vibrio coralliilyticus* and menthol
had strong and significant impacts on transcriptome-wide gene expression
that are independent or additive in their effects, but not interactive.
*Vibrio* exposure had the strongest impact (4,164 DE genes) followed by
menthol treatment (1,114 DE genes) and then the additive combinations of
*Vibrio* and menthol (472 DE genes). KEGG enrichment analyses identified
11 pathways - involved in immunity (5), transport and catabolism (4) and
cell growth and death (2) - that were enriched due to both *Vibrio*
and/or menthol exposure. Seven pathways were enriched for both *Vibrio*
and menthol (complement and coagulation cascades, chemokine signaling,
endocytosis, lysosome, peroxisome, apoptosis, P53 signaling), and four
pathways were enriched for *Vibrio* alone (antigen processing and
presentation, leukocyte transendothelial migration, NOD-like receptor
signaling, phagosome). However, over-representation of DE genes was only
significant for *Vibrio* exposure at a complement and coagulation
cascade pathway. Pathway level patterns in gene expression are discussed
below.

**Immune System Response**

Amongst the immune system pathways, there was strong evidence that the
complement and coagulation cascade was responding to both *Vibrio* and
menthol exposure, whereas NOD/TLR pathway, chemokine and antigen
processing were initiated primarily by *Vibrio* exposure. The
stimulation of complement and coagulation cascade pathway and NOD/TLR
pathway indicates that bacterial immune challenge by *Vibrio* involves
two of the three primary innate immune pathways in invertebrates; we did
not find significant evidence for stimulation of the prophenoloxidase
(PPO) activating system (i.e. melanization), which has been shown to be
up-regulated in some hard and soft coral immune responses
\[Palmer2011\].

*<span class="underline">Complement and Coagulation Cascade</span>*

Overall, these gene expression patterns of gene expression in the
complement and coagulation cascade indicate that coagulation is
initiated by by *Vibrio* and menthol exposure, whereas the complement
alternative pathway is initiated primarily by *Vibrio* exposure.

Complement and coagulation cascade pathway had 10 highly expressed DE
genes; four DE genes in the coagulation cascade and six DE genes in the
complement cascade. Three out of the four coagulation genes were
up-regulated due to *Vibrio* and menthol exposure \[(von Willebrand
factor (VWF), alpha-2-macroglobulin (A2MG), tissue factor pathway
inhibitor (TFPL1)\] while coagulation factor XIII B polypeptide (F13B)
was up-regulated for *Vibrio* exposure only. Five out of the six DE
genes in the complement pathway were up-regulated for *Vibrio* only
(MCP, DAF, CFAB, C3, CFAH) while PAI2 was down-regulated for both
*Vibrio* and menthol.

The coagulation pathway is a balance between procoagulant factors that
close wounds and prevent fluid loss, and anticoagulant factors that
prevent coagulation in surrounding healthy tissue and dissolve clots
after healing has occurred \[Palta2014\]. The release of
transglutaminase agglutinates proteins into a gel to entrap pathogens
and to seal wounds following injury \[Cerenius2010\], while plasmin and
its regulators are involved in clot dissolution \[Hoover-Plow2010\]. The
complement alternative pathway opsonizes pathogens through nonspecific
covalent bonding of C3b, relying on complement regulators like DAF, MCP,
and CFAH to identify and protect healthy host cells. Opsonization by C3b
initiates removal by phagocytosis \[Cerenius2010\],\[Ferriera2010\],
\[Schmidt2016\].

Out of the four highly expressed DE genes that differed due to *Vibrio*
and menthol exposure, three genes (VFW, A2MG, TFPL1) have previously
been associated with immune challenge in anemones
\[Lanetty2006\],\[Stewart2017\] and corals
\[Oren2009\],\[Libro2013\],\[Libro2016\]. Von Willebrand factors (VWF)
have also been observed to be up-regulated in symbiotic *Aiptasia
\[Lanetty2006\]*. To our knowledge, our data are the first to associate
the up-regulation of plasminogen activator inhibitor 2 with either
immune challenge or symbioses.

Von Willebrand factor (VWF) is involved in cell adhesion and collagen
binding and in vertebrates facilitates platelet adhesion to damaged
blood vessels \[Ruggeri2007\]. VWF has been associated with allogeneic
rejection, pathogen exposure, and symbiotic state in cnidarians. Oren et
al (2009) observed up-regulation of VWF during allogeneic rejection in
the coral *Stylophora pistillata*. Libro et al (2013) observed
up-regulation of VWF in response to White Band Disease infection in the
coral *Acropora cervicornis*. Lanetty et al (2006) observed
up-regulation of VWF in symbiotic versus aposymbiotic forms of the
anemone *Anthopleura elegantissima.* Our data indicate that VWF is
up-regulated due to menthol induced bleaching and thus the difference
between our data and Lanetty et al. may reflect differences in VWF
expression between the initial breakdown in symbioses versus a stable
aposymbiotic state. Overall, our data coupled with published data
indicate that up-regulation of VWF may be a hallmark of anthozoan
immunity, allorecognition, and symbioses.

Alpha-2-macroglobulin (A2M) binds peptides including a wide range of
proteinases \[Borth1992\]. The ability to bind the serine protease
thrombin gives A2M anticoagulant properties \[Mitchell1991\], while the
ability to inhibit protein C and protein S gives it procoagulant
properties \[Cvirn2002\]. Many pathogen virulence factors act as
proteases, and A2M's ability to inhibit proteases protects the host from
these virulence factors \[Armstrong1999\]. A2M has been associated with
pathogen exposure in corals \[Libro2013\] and wound healing in anemones
\[Stewart2017\], but not symbioses. Libro et al (2013) observed
up-regulation of A2M in response to White Band Disease infection in
*Acropora cervicornis* and Stewart et al. (2017) identified that A2M was
among the highest up-regulated genes in response to wound healing in the
anemone *Calliactis polypus*. Our results demonstrate it is also
upregulated during the breakdown of symbioses.

Tissue factor pathway inhibitor (TFPI) inhibits coagulation by
inhibition of factor Xa and factor VIIa/tissue factor \[Broze2013\], and
has been associated with coral immunity \[^Libro2016\] and oxidative
stress in anemones \[Richier2008\], but not the breakdown of symbioses.
Libro et al (2013) observed up-regulation of TFPI in response to white
band disease infection in *Acropora cervicornis*, and Richier et al
(2008) observed down-regulation of TFPI in response to UV and thermal
stress.

Plasminogen activator inhibitor 2 (PAI2) is a serine protease inhibitor
associated with inhibition of fibrinolysis \[Stump1986\] and negative
regulation of apoptosis \[Dickinson1995\]. PAI2 inhibits the
fibrinolytic urokinase plasminogen activator (uPA)
\[Cochran2009\],\[Stump1986\]. Down-regulation of PAI2 for both *Vibrio*
and menthol exposure suggests a net anticoagulant effect. Serine
protease inhibitors have not been associated with anthozoan immunity or
symbioses, but both up and down-regulation of serine protease inhibitors
have been reported in other invertebrates like the silkworm *Bombyx
mori* in response to bacterial challenge. \[Zhao2012\]

Overall, the patterns of DE in the coagulation genes (A2M, TFPI, PAI2)
suggest that coagulation is inhibited. While the coagulation pathway is
typically involved in wound healing, another possible explanation for
the differential expression of genes within coagulation due to menthol
exposure is the breakdown of coagulants within the symbiosome. Kanzjian
et al (2008) observed high levels of coagulants in isolated symbiosomes
of the coral *Zoanthus robustus*. This observation was secondary to
their study, but the high levels of coagulants within the symbiosomes
prevented the use of Percoll gradients to purify the symbiosomes. The
inhibition we observed may be involved in the breakdown of the
symbiosome.

Within the complement cascade, the six highly expressed DE genes that
were up-regulated for *Vibrio* (C3, factor B, coagulation factor XII B
polypeptide, membrane cofactor protein, decay accelerating factor,
factor H) indicate that complement is key to responding to bacterial
pathogen infection. Complement C3, plays a central role in both the
classical and alternative complement pathways, and is activated by C3
convertase and then opsonizes pathogens leading to phagocytosis.
Up-regulation of C3 has been observed in the coral *Acropora millepora*
treated with bacterium *Alteromonas* \[Brown2013\] and in WBD-infected
*Acropora cervicornis* \[Libro2016\].

Component factor B (CFAB) is part of the alternative pathway and is
involved in the generation of C3 convertase and downstream cleavage of
C3 into its active components. Poole et al (2016) identified two
variants of Factor B in *Aiptasia*, one of which was up-regulated in
response to both onset of symbiosis and treatment with *Serratia
marcescens*. Up-regulation of coagulation factor XII B chain (F13B), is
associated with blood coagulation and hemostasis in vertebrates, and has
also been observed in diseased *Acropora cervicornis*
\[LibroThesis2014\].

In the complement pathway, the classical and lectin pathways require
specific recognition molecules for initiation, but in the alternative
pathway, C3b is deposited on all cells (host as well as pathogenic)
exposed to activated complement \[Ferreira2010\]. Three remaining highly
expressed DE genes within complement (membrane cofactor protein,
complement decay-accelerating factor, factor H) are all involved in
protecting host tissues from attack by the less specific alternative
complement pathway \[Elvington2016\], \[Ferreira2010\]. Up-regulation of
Membrane cofactor protein (MCP/cd46), complement decay-accelerating
factor (DAF/cd55), and Component factor H in *Aiptasia* would limit C3b
deposition on healthy *Aiptasia* cells \[Elvington2016\],
\[Ferreira2010\]. Neither DAF, MCP, or CFAH have previously been
associated with anthozoan immunity.

<span class="underline">NOD/TLR pathway.</span>

NOD/Toll-like receptor pathway had six highly DE genes \[3 up and 3
down\] due to Vibrio exposure. Myeloid differentiation primary response
protein (MyD88), TNF receptor-associated factor 3 (TRAF3) and Bcl-2-like
1 apoptosis regulator Bcl-X (Bcl-2) were up-regulated, while TNF
receptor-associated factor 2 (TRAF2), receptor-interacting
serine/threonine-protein kinase 2 (RIPK2), and calcium-sensing receptor
(CASR) were down-regulated. MyD88, TRAF3, and RIPK2 are key regulators
of the NOD and TLR pathways that lead to NF-kappa-β activation, cytokine
secretion and the inflammatory response, while Bcl-2 inhibits caspases
and suppresses apoptosis. TRAF2 regulates activation of NF-kappa-β and
JNK and CASR regulates calcium homeostasis.

Out of the six DE genes in the NOD/Toll-like receptor pathway, only
MyD88 has previously been observed to be DE in cnidarians due to immune
exposure. Libro et al (2013) observed up-regulation of MyD88 in
WBD-infected *Acropora cervicornis*. In humans \[Wang2000\], mouse,
\[Dequine2014\], and fly \[Horng2001\], stimulation of toll-like
receptors (TLRs) causes MyD88 to associate with the intracellular domain
of the TLR leading to downstream signaling of NF-kappa-β via IRAK and
TRAF and production of proinflammatory cytokines \[Akira2006\].

TLR activation of MyD88 has been demonstrated in the anemone
*Nematostella vectensis* in a reporter gene assay where *Nematostella*
TIR domain of TLR activated human MyD88. Our results indicate that MyD88
interacts with TRAF3 but not IRAK, which is supported by a MyD88
knockdown study by Franzenburg et al (2012) in the hydrozoan<sup>-</sup>
*Hydra vulgaris*, which resulted in the down-regulation of TRAF3 but not
IRAK.

The expression patterns of Bcl-2, RIPK2, and CASR suggest that they are
acting to prevent apoptosis in *Aiptasia* exposed to *Vibrio*. The
B-cell lymphoma 2 (Bcl-2) family of apoptosis-regulating proteins
includes both pro and anti-apoptotic members. The Aiptasia Bcl-2 homolog
up-regulated in this study was the anti-apoptotic Bcl-W. Dunn et al
(2006) identified Bcl-2 family members including Bcl-W in *Aiptasia*
cDNA. Ainsworth et al (2015) identified up-regulation of the
pro-apoptotic Bcl-2 family member Bak in *Acropora hyacinthus* tissues
exhibiting white syndrome and Pernice et al (2011) proposed that
up-regulation of Bcl-2 is a protective response to heat-stress-induced
apoptotic activity in *Acropora millepora*. Down-regulation of RIPK2 and
CASR also suggests an anti-apoptotic role in *Aiptasia* exposed to
*Vibrio*. RIPK2 is a protein kinase that activates NF-kappa-β and
induces cell death \[McCarthy1998\]. CASR is involved in the JNK cascade
and apoptotic process, and up-regulation of CASR leading to apoptosis
has been reported in rat myocytes exposed to LPS. To our knowledge we
are the first to report differential expression of RIPK2 and CASR in
Cnidaria

TRAF2 transduces TNF signaling to activate NF-kappa B, activate JNK, and
suppress apoptosis \[Xia2002\]. TRAF2 had not been associated with
cnidarian immunity, but had been shown to be DE due to injury in Hydra
(Pernice et al 2011, Wegner et al 2014). Down-regulation of TRAF2
suggests that it is working counter to Bcl-2, RIPK2, and CASR to promote
apoptosis in *Aiptasia* exposed to *Vibrio*

Even though Aiptasia shows strong evidence for TLR pathway activation,
no *Aiptasia* TLRs meeting our annotation criteria (best hit, e-value \<
1e-10, coverage \> 50%). Two *Aiptasia* genes (KXJ18603.1, KXJ08560.1)
annotated as a relaxin receptor 2 (RXFP2) and outer membrane protein
OprM (OPRM) were up-regulated for *Vibrio* and had had blast hits for a
TLR with an e-value \< 1e-10 and coverage greater than 50%. While it is
possible that these two genes represent TLRs, more data would be needed
to confirm their putative functions.

What can we say about NOD/TLR in general…. ?

<span class="underline">Chemokine and Antigen Processing</span>

Chemokine and antigen processing pathways were also activated by
*Vibrio* exposure. Chemokine pathway had three highly DE genes - signal
transducer and activator of transcription 1-alpha/beta (STAT1) was
up-regulated while C-X-C chemokine receptor type 4 (CXCR4), and guanine
nucleotide-binding protein subunit beta-4 (GBB4) were down-regulated.
STAT1 mediates cellular responses to interferons (IFNs), cytokines and
other growth factors \[Ramana2000\], and up-regulation of STAT in
response to bacterial exposure has been reported in a number of
invertebrates including *Anopheles gambiae* (mosquito)
\[Barillas-Mury1999\], *Drosophila*, and *Fenneropenaeus chinensis*
(Chinese white shrimp). Sinkovics (2015) proposed a Cnidarian origin of
STAT based on genomic studies on *Nematostella vectensis*, but this has
not been confirmed by expression analysis.

CXCR4 transduces a signal by increasing intracellular calcium ion levels
and enhancing MAPK1/MAPK3 activation \[UniProt\] and GBB4 modulates or
transduces various transmembrane signaling systems. \[UniProt\]. To our
knowledge, none of these genes have been associated with cnidarian
immunity before.

Two antigen processing genes were highly DE; Proteasome activator
subunit 2 (PSME2/PA28 beta) was up-regulated for Vibrio, and regulator
factor X-associated ankyrin-containing protein (RFXK) was down-regulated
for menthol. What do they do and have they been associated before.

**Transport and catabolism**

Some summary sentence about relative importance peri, phago, lyso and
whether vibrio and or menthol

There were also more genes highly down-regulated than up-regulated for
*Vibrio* within the transport and catabolism peroxisome, endocytosis,
and lysosome pathways.

<span class="underline">Peroxisome</span>

Peroxisome pathway had 7 highly expressed DE genes; one gene was up for
both Vibrio and menthol exposure (PAOX), 3 genes were down for *Vibrio*
only (SOX, LCFB, DHRS4), one gene was down for both (GNPAT), one gene
was up for menthol only (EASC) and one gene was down for menthol only
(BAAT).

Peroxisomes are involved in resolution of bacterial infection through
innate immune signaling \[DiCara2017\]. Which genes are really important
in process. What immune signaling?

N1-acetylpolyamine oxidase (PAOX) is a polyamine oxidase that oxidizes
N1-acetylspermine to spermidine and hydrogen peroxide. PAOX has been
previously observed to be up-regulated in disease infected *Acropora
cervicornis* \[^Libro2016\], and up-regulated in aposymbiotic anemones
*Anemonia viridis* Ganot et al (2011). WHAT IS A KERN SET? .

Dihydroxyacetone phosphate acyltransferase (GNPAT/DHAPAT) is involved in
production of lipid molecules called plasmalogens \[Braverman2012\].
Lehnert et al (2014) observed up-regulation of GNPAT/DHAPAT in symbiotic
*Aiptasia*. To our knowledge DE of GNPAT has not been reported in
cnidarian following immune challenge.

<span class="underline">Endocytosis</span>

Endocytosis pathway had six highly expressed DE genes; three genes were
up-regulated for *Vibrio* (HRS, HSE1, CLH), 2 genes were down-regulated
for *Vibrio* (VPS4, PLD2), and one gene was down for menthol (JUNO).
DESCRIBE A LITTLE ABOUT ENDOCYTOSIS PATHWAY IN GENERAL

HRS plays a role in endosome membrane invagination and formation of
multivesicular bodies, HSE1 is a component of the sorting receptor
complex for ubiquitinated cargo proteins at the multivesicular body
\[UniProt\], and clathrin heavy chain (CLH) is the major protein of the
polyhedral coat of coated pits and vesicles \[UniProt\]. Up-regulation
of CLH has been observed in WBD-infected Acropora cervicornis
\[Libro2016\]. None of the other genes have been associated with
anthozoan immunity or symbioses.

<span class="underline">Lysosome</span>

Lysosome had four highly DE genes; . AP-3 complex subunit delta (AP3D)
was up-regulated for Vibrio, N-acetylgalactosamine-6-sulfatase (GALNS)
was down-regulated for Vibrio, and lysophospholipase III (PAG15) and
beta-glucuronidase (BGLR) were both down-regulated for menthol. What
does lysosome do… in general.

AP3D facilitates formation of vesicles from the Golgi membrane and may
be involved in trafficking to lysosomes. \[UniProt\].. OTHER GENES.
EVIDENCE IN CNIDARIANS.

**Apoptosis - programmed cell death**

Apoptosis pathway had 5 highly expressed DE genes; two caspases were
up-regulated for Vibrio (CASP7, CASP9), (TBA) was down-regulated for
Vibrio (TBA), (P53) was up-regulated for menthol, and (BIRC5) was
down-regulated for menthol. SOMETHING ABOUT APOPTOSIS IN IMMUNITY OF
BLEACHING IN GENERAL.

Caspases are key initiators of apoptosis \[UniProt\]. Up-regulation of
CASP7 and CASP9 have not been previously reported in cnidarians, but
up-regulation of caspase-3 was documented in WBD-infected *Acropora
cervicornis*. \[Libro2016\]. Tubulin alpha (TBA) was down-regulated due
to Vibrio exposure in Aiptasia. Down-regulation of tubulin beta has been
observed for WBD-infected *Acropora cervicornis as well*\[Libro2016\]

T

HERE IS WHERE WE CAN TRY TO WRAP UP.
